### PR TITLE
 feat: png data -> pixel data

### PR DESCRIPTION
**Pull request by thomas192**
- [x] issue #8 
- [x] follows contribution [guide](https://github.com/keep-starknet-strange/art-peace/blob/main/CONTRIBUTING.md)
- [ ] code change includes tests
- [ ] breaking change

Hey, I wanted to get this done a couple days ago, but got pulled into a few things.
I wanted to write tests but I'm not sure where to do it, so I tested it manually with curl and an 8x1 image.
Let me know if this is what was indeed expected :)

### CODE REVIEW

File: backend/routes/templates.go [hunk 1]
```diff
@@ -35,10 +40,58 @@ func hashTemplateImage(pixelData []byte) string {
...
-	// TODO: Convert image data to pixel data using approximation
-	//       Output should be a byte array with color indexes
-	return []byte{0, 1, 1, 2, 2, 3}
+func imageToPixelData(imageData []byte) ([]byte, error) {
+	img, _, err := image.Decode(bytes.NewReader(imageData))
+	if err != nil {
+		return nil, err
+	}
+
+	palette := []color.Color{
```
Comment (b-j-roberts) [inline backend/routes/templates.go:49]: Can you load the color palette from `core.ArtPeaceBackend.CanvasConfig.Colors` and dynamically generate this array from that data?

**Review by b-j-roberts — CHANGES_REQUESTED** [review]
No worries, great work!

Yeah, there is no testing framework for the backend at the moment. And building one is kinda low priority for now, since the stakes are so low. So all good there!

**Comment by b-j-roberts** [comment]
Hey @thomas192 , wil you be able to address the comment soon? If not, no worries, I can just merge what you have so far and have someone else work on that part.

**Comment by thomas192** [comment]
> Hey @thomas192 , wil you be able to address the comment soon? If not, no worries, I can just merge what you have so far and have someone else work on that part.

Hello Brandon, sorry my attention is elsewhere at the moment. I should have let you know.

**Comment by Gift-Naomi** [comment]
Hi @b-j-roberts can you please assign me to work on this.

**Comment by Immanuelolivia1** [comment]
Hi @b-j-roberts I can handle this

**Comment by b-j-roberts** [comment]
> Hello Brandon, sorry my attention is elsewhere at the moment. I should have let you know.

No worries. I can assign the rest to someone else and will reward you for the work here after it's merged.

**Pull request merged**